firmware/projects/DemoProject/platforms/stm32f767/cubemx/CMakeLists.txt

```diff
@@ -3,7 +3,22 @@
 
 message(STATUS "Building CubeMX objects with the auto-generated Makefile")
 if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/Makefile")
-	message(FATAL_ERROR "No auto-generated Makefile was found. Open the .ioc file in STM32CubeMX and click [Generate Code]")
+	find_program(STM32CubeMX_PATH STM32CubeMX.exe)
+
+	if(NOT STM32CubeMX_PATH)
+		message(FATAL_ERROR "STM32CubeMX.exe not found in PATH")
```

Comment:
```suggestion
		message(FATAL_ERROR "STM32CubeMX.exe not found in PATH. Open the .ioc file in STM32CubeMX and click [Generate Code]")
```

Reply:
I don't think this will work,. CMake will report FATAL_ERROR when it reaches the stage where it looks for CubeMX in the path, regardless of whether the user has manually clicked [generate code].

Since the user needs CubeMX installed for manual and automatic generation, I suggest we enforce it to be on the path

Reply:
Ahh good point. The only issue is that Unix based devices will not have the `.exe` extension.

Reply:
If you manually generate, the make file will exist and it will not attempt to generate. This might be a problem however. We should maybe add a condition that the `board_config.ioc` file has not changed

Reply:
> If you manually generate, the makefile will exist and it will not attempt to generate. This might be a problem however. We should maybe add a condition that the `board_config.ioc` file has not changed

Does this mean that it should only generate if the .ioc file has changed? Also regarding the comment above, should I implement the suggestion or keep it as is to enforce it to be on path?

Reply:
Yes it should only regen if the ioc changes. Otherwise the build process will be very long for every build.
If we were using Make instead of CMake, then the recipe
```
<any cubemx generated file>: board_config.ioc
    <Java generate call>
```
Would regenerate only when the build_config.ioc is updated after the cubemx file is generated. 

I think you can do a similar thing with CMake

Reply:
And it should require STM32cubemx on path

Reply:
The only way I can think of doing it is storing the timestamp of the last build and reading it once we try to build again and seeing if it matches. Would that be a good solution or is there a better way to do this?

Reply:
I think the best thing to do at this point is to get this working on a Unix machine. The team raspi would be nice and it is always accessible. CMake allows you to check if you are on a Windows OS or a Unix based OS so if there are major differences you could branch on this condition.

Reply:
> The only way I can think of doing it is storing the timestamp of the last build and reading it once we try to build again and seeing if it matches. Would that be a good solution or is there a better way to do this?

I'm pretty sure CMake has this functionality built in

Reply:
> I think the best thing to do at this point is to get this working on a Unix machine. The team raspi would be nice and it is always accessible. CMake allows you to check if you are on a Windows OS or a Unix based OS so if there are major differences you could branch on this condition.

how would the command differ if we are trying to run it in Unix? How can I even test that, I do not own any non windows machines?

Reply:
resolved by my modifications